qt/src/hocr/HOCRDocument.cc

```diff
@@ -82,6 +82,59 @@ bool HOCRDocument::editItemAttribute(const QModelIndex& index, const QString& na
 	return true;
 }
 
+// Might be more logical to accept oldParent and oldRow as parameters rather than itemIndex
+// (since we need them anyway), if all our callers are likely to know them. Swap does; drag&drop might not.
+QModelIndex HOCRDocument::moveItem(const QModelIndex& itemIndex, const QModelIndex& newParent, int newRow, bool page) {
```

Comment:
You can detect whether it is a page via `item->itemClass()`

Reply:
Thanks. Done.